test/components/guideLineLayerTests.ts

```diff
@@ -0,0 +1,401 @@
+      let linearScale = new Plottable.Scales.Linear();
+      linearScale.domain([0, 1]);
+      linearScale.range([0, 100]);
+
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      let value = 0.5;
+      gll.value(value);
+      let setPosition = -999;
+      gll.pixelPosition(setPosition);
+      let expectedPosition = linearScale.scale(value);
+      gll.scale(linearScale);
+      assert.strictEqual(gll.pixelPosition(), expectedPosition, "setting the scale updates the pixel position");
+      assert.notStrictEqual(gll.pixelPosition(), setPosition, "originally-set pixel position was overridden");
+
+      let linearScaleB = new Plottable.Scales.Linear();
+      linearScaleB.domain([0, 1]);
+      linearScaleB.range([0, 200]);
+      let expectedPositionB = linearScaleB.scale(value);
+      gll.scale(linearScaleB);
+      assert.strictEqual(gll.pixelPosition(), expectedPositionB, "changing the scale updates the pixel position");
+      assert.strictEqual((<any>linearScale)._callbacks.size, 0, "callback was removed from the previous Scale");
+    });
+
+    it("changing the scale updates value() if pixelPosition() is set but value() is not", () => {
+      let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+      let position = 50;
+      gll.pixelPosition(position);
+
+      let linearScale = new Plottable.Scales.Linear();
+      linearScale.domain([0, 1]);
+      linearScale.range([0, 100]);
+
+      gll.scale(linearScale);
+      let expectedValue = linearScale.invert(position);
+      assert.strictEqual(gll.value(), expectedValue, "value was updated to match the set pixel position");
+    });
+  });
+
+  it("rejects invalid orientations", () => {
+    assert.doesNotThrow(() => new Plottable.Components.GuideLineLayer<number>("vertical"), Error, "accepts \"vertical\"");
+    assert.doesNotThrow(() => new Plottable.Components.GuideLineLayer<number>("horizontal"), Error, "accepts \"horizontal\"");
+    assert.throws(() => new Plottable.Components.GuideLineLayer<number>("blargh"), Error);
+  });
+
+  describe("rendering (vertical)", () => {
+    let SVG_WIDTH = 400;
+    let SVG_HEIGHT = 300;
+
+    it("has an effective size of 0, but will occupy all offered space", () => {
+      let gll = new Plottable.Components.GuideLineLayer<void>("vertical");
+      let request = gll.requestedSpace(SVG_WIDTH, SVG_HEIGHT);
+      TestMethods.verifySpaceRequest(request, 0, 0, "does not request any space");
+      assert.isTrue(gll.fixedWidth(), "fixed width");
+      assert.isTrue(gll.fixedHeight(), "fixed height");
+
+      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+      gll.anchor(svg);
+      gll.computeLayout({x: 0, y: 0}, SVG_WIDTH, SVG_HEIGHT);
+      assert.strictEqual(gll.width(), SVG_WIDTH, "accepted all offered width");
+      assert.strictEqual(gll.height(), SVG_HEIGHT, "accepted all offered height");
+      svg.remove();
+    });
```

Comment:
No, since the combination of "space-accepted" and "space used" is critical to the `GuideLineLayer`'s behavior during layout; therefore it makes sense to test them together.